src/tools/Avalonia.Generators/Common/GlobPatternGroup.cs

```diff
@@ -1,17 +1,21 @@
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Linq;
 using Avalonia.Generators.Common.Domain;
 
 namespace Avalonia.Generators.Common;
 
 internal class GlobPatternGroup : IGlobPattern
 {
-    private readonly GlobPattern[] _patterns;
+    private readonly ImmutableArray<GlobPattern> _patterns;
 
     public GlobPatternGroup(IEnumerable<string> patterns) =>
-        _patterns = patterns
-            .Select(pattern => new GlobPattern(pattern))
-            .ToArray();
+        _patterns = patterns.Select(p => new GlobPattern(p)).ToImmutableArray();
 
     public bool Matches(string str) => _patterns.Any(pattern => pattern.Matches(str));
+
+    public bool Equals(IGlobPattern other) => _patterns.Any(pattern => pattern.Equals(other));
```

Comment:
This doesn't match `GetHashCode`, this should be `other is GlobalPatternGroup group && _patterns.SequenceEqual(group._patterns)`